src/appengine/handlers/viewer.py

```diff
@@ -45,13 +45,20 @@ def get(self):
       if not access.has_access():
         raise helpers.AccessDeniedException()
 
-    if blobs.get_blob_size(key) > MAX_ALLOWED_CONTENT_SIZE:
+    blob_size = blobs.get_blob_size(key)
+    if blob_size > MAX_ALLOWED_CONTENT_SIZE:
       raise helpers.EarlyExitException('Content exceeds max allowed size.', 400)
 
-    try:
-      content = unicode(blobs.read_key(key), errors='replace')
-    except Exception:
-      raise helpers.EarlyExitException('Failed to read content.', 400)
+    # TODO(mbarbella): Workaround for an issue in the Cloud Storage API. Remove
+    # once it is fixed properly upstream:
+    # https://github.com/googleapis/google-cloud-python/issues/6572
+    if blob_size:
+      try:
+        content = unicode(blobs.read_key(key), errors='replace')
```

Comment:
maybe remove this unicode thing along the way and fix https://bugs.chromium.org/p/chromium/issues/detail?id=920160 :) stage to make sure it works.

Reply:
SGTM.

Reply:
Seems to cause issues later in template rendering, which I guess explains why we had the hack initially. I'm going to leave it out of this change.